docs/root/start/simple_example.rst

```diff
@@ -0,0 +1,259 @@
+.. _start_simple:
+
+Simple Example
+==============
+
+This is perhaps the most minimal Envoy configuration which can be used to validate basic plain HTTP
+proxying is available in :repo:`configs/google_com_proxy.v2.yaml`. This is not
+intended to represent a realistic Envoy deployment.
+
+Quick Start
+-----------
+
+This Quick Start runs the example quickly using files in the Envoy repo. The sections below explain
+the configuation file and execution steps in detail.
+
+Copy both :repo:`configs/Dockerfile` and
+:repo:`configs/google_com_proxy.v2.yaml` to the same directory on your local
+disk. Then, build and run the Dockerfile, and test out Envoy by sending a
+request to port 10000::
+
+  $ docker build -t envoy-google-test:v1 .
+  $ docker run -d -p 10000:10000 envoy-google-test:v1
+  $ curl -v localhost:10000
+
+The Dockerfile will build a container containing the latest version of Envoy,
+and copy a basic Envoy configuration into the container. This basic
+configuration tells Envoy to route incoming requests to \*.google.com.
+
+
+Simple Configuration
+--------------------
+
+Envoy can be configured using a single YAML file passed in as an argument on the command line.
+
+Below is the `google_com_proxy.v2.yaml` file.
+
+.. code-block:: yaml
+
+  admin:
+    access_log_path: /tmp/admin_access.log
+    address:
+      socket_address: { address: 0.0.0.0, port_value: 9901 }
+    static_resources:
+      listeners:
+      - name: listener_0
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 10000 }
+        filter_chains:
+        - filters:
+          - name: envoy.http_connection_manager
+            config:
+              stat_prefix: ingress_http
+              route_config:
+                name: local_route
+                virtual_hosts:
+                - name: local_service
+                  domains: ["*"]
+                  routes:
+                  - match: { prefix: "/" }
+                    route: { host_rewrite: www.google.com, cluster: service_google }
+              http_filters:
+              - name: envoy.router
+      clusters:
+      - name: service_google
+        connect_timeout: 0.25s
+        type: LOGICAL_DNS
+        # Comment out the following line to test on v6 networks
+        dns_lookup_family: V4_ONLY
+        lb_policy: ROUND_ROBIN
+        hosts: [{ socket_address: { address: google.com, port_value: 443 }}]
+        tls_context: { sni: www.google.com }
+
+
+Let's go through each section in detail.
+
+.. code-block:: yaml
+
+  admin:
+    access_log_path: /tmp/admin_access.log
+    address:
+      socket_address: { address: 0.0.0.0, port_value: 9901 }
+
+..
+
+  The :ref:`admin message <envoy_api_msg_config.bootstrap.v2.Admin>` is required to configure
+  the administration server.
+
```

Comment:
I tried it without the indentation below the blocks and it was harder to follow. I also tried having the blocks be below the text and it was worse. I think the indentation of the description below each YAML block makes it easier to read. Unfortunately there is no way using Sphinx (that I could find) to have the YAML code blocks be properly indented (which is why I included the whole file at the beginning). If I would present the code blocks properly indented, the indentation of the description below the blocks would look nicer.

Reply:
The code blocks in the example here: http://www.sphinx-doc.org/en/stable/markup/code.html show indentation (you can click on show source to see how it's done).

Now that you explained it, I see why you did what you did. It struck me as odd, though, initially. I don't feel strongly about it, just stuck out.

Reply:
@richarddli I was looking to indent *within* the code block, so that if you looked at all of the code blocks together, you could see where you were, not to indent the code block (as shown in your link). There is a "dedent" option in Sphinx, but not a corresponding "indent" option.

Reply:
Agree this looks a bit strange, but not sure how to make it better right now. We can always iterate.